src/js/components/SidebarComponent.jsx

```diff
@@ -0,0 +1,114 @@
+var Link = require("react-router").Link;
+var React = require("react/addons");
+
+var AppListLabelsFilterComponent =
+  require("../components/AppListLabelsFilterComponent");
+var AppListStatusFilterComponent =
+  require("../components/AppListStatusFilterComponent");
+
+var SidebarComponent = React.createClass({
+  displayName: "SidebarComponent",
+
+  contextTypes: {
+    router: React.PropTypes.func
+  },
+
+  propTypes: {
+    groupId: React.PropTypes.string.isRequired,
+    onChange: React.PropTypes.func.isRequired
+  },
+
+  getInitialState: function () {
+    return {
+      filters: {
+        filterLabels: [],
+        filterStatus: []
+      }
+    };
+  },
+
+  updateFilterLabels: function (filterLabels) {
+    var state = this.state;
+
+    var filters = Object.assign({}, state.filters, {
+      filterLabels: filterLabels
+    });
+
+    this.setState({
+      filters: filters
+    }, this.props.onChange(filters));
+  },
+
+  updateFilterStatus: function (filterStatus) {
```

Comment:
Couldn't we have a single handler for all filters? I see a lot of unnecessary repetition.